### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by lgalabru**
Add a quick intro, roadmap, and small tutorial for interacting with the testnet

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,243 @@
...
+- Open-membership means that any host on the Internet can join the blockchain and independently calculate the same full replica as all other peers.
+- Non-enumerable means that the set of peers that are producing the blocks don’t know about one another — they don’t know their identities, or even how many exist and are online.  They are indistinguishable.
+
+## Roadmap
+
+- [x] [SIP 001: Burn Election](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-001-burn-election.md)
+- [x] [SIP 002: Clarity, a language for predictable smart contracts](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-002-smart-contract-language.md)
+- [x] [SIP 004: Cryptographic Committment to Materialized Views](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-004-materialized-view.md)
+- [x] SIP 005: Blocks, Transactions, and Accounts
```
Comment (jcnelson) [inline README.md]: I think SIP 005 is ready, if you all want to take a look at the SIP 005 PR.  Then we can link it here :pray:

File: README.md [hunk 2]
```diff
@@ -1,2 +1,243 @@
-# blockstack-ng
-Reference implementation of the Stacks blockchain v2
+# Stacks 2.0
+
+Reference implementation of https://blockstack.org/whitepaper.pdf in Rust.
```
Comment (jcnelson) [inline README.md]: "is in Rust."

File: README.md [hunk 3]
```diff
@@ -1,2 +1,243 @@
...
+## Roadmap
+
+- [x] [SIP 001: Burn Election](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-001-burn-election.md)
+- [x] [SIP 002: Clarity, a language for predictable smart contracts](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-002-smart-contract-language.md)
+- [x] [SIP 004: Cryptographic Committment to Materialized Views](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-004-materialized-view.md)
+- [x] SIP 005: Blocks, Transactions, and Accounts
+- [ ] [SIP 003: Peer Network](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-003-peer-network.md) (Q1 2020)
+- [ ] SIP 006: Clarity Execution Cost Assessment (Q1 2020)
+
+Stacks improvement proposals (SIP) are documented design proposals aimed at improving the Stacks blockchain. They should contain concise technical specifications of features or standards and the rationale behind it. SIPs are intended to be the primary mechanism of proposing new features, for collecting of community input on an issue and for documenting design decisions.
```
Comment (jcnelson) [inline README.md]: "Stacks improvement proposals (SIPs) are aimed at describing the implementation of the Stacks blockchain, as well as proposing improvements."

Comment (jcnelson) [reply README.md]: "SIPs are intended to be the primary medium for proposing new features, for collecting community input on a system-wide issue, and for documenting design decisions."

File: README.md [hunk 4]
```diff
@@ -1,2 +1,243 @@
...
+
+See [SIP 000](https://github.com/blockstack/blockstack-core/blob/develop/sip/sip-000-stacks-improvement-proposal-process.md) for more details.
+
+### Testnet versions
+
+- [x] **Local Testnet** is a developer local setup, mono-node, assembling SIP 001, SIP 002, SIP 004 and SIP 005. With this version, developers can run Stack 2.0 on their development machines, but also write, execute and tests smart contracts. See the instructions below for more details.
```
Comment (jcnelson) [inline README.md]: "...developers can not only run Stacks 2.0 on their development machines, but also write, execute, and test smart contracts."

File: README.md [hunk 5]
```diff
@@ -1,2 +1,243 @@
...
+
+- [ ] **Open Testnet** is the incoming version of our testnet, that we're intending to ship in Q1 2020. This testnet will ship with SIP 003 and be an open membership network, where participants will be able to observe and experience the consensus running at scale.
```
Comment (jcnelson) [inline README.md]: "***Open Testnet** is the upcoming version of our public testnet, that we're anticipating will ship in Q1 2020.  This testnet will ship with SIP 003, and will be an open-membership public network, where participants will be able to validate and participate in mining testnet blocks."

File: README.md [hunk 6]
```diff
@@ -1,2 +1,243 @@
...
+
+From there, you can clone this repository:
+
+```bash
+git clone https://github.com/blockstack/blockstack-core.git
+
+cd blockstack-core
+```
+
+Then build the project:ca
```
Comment (jcnelson) [inline README.md]: Typo

File: README.md [hunk 7]
```diff
@@ -1,2 +1,243 @@
...
+cargo run --bin blockstack-cli publish --help
+```
+
+With the following arguments:
+
+```bash
+cargo run --bin blockstack-cli publish b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001 0 0 kv-store ./kv-store.clar --testnet
+```
+
+This command will output the **binary format** of the transaction. In our case, we want to pipe this output and dump it to a file that will be used later in this step by step.
```
Comment (jcnelson) [inline README.md]: "...will be used later in this tutorial."

File: README.md [hunk 8]
```diff
@@ -1,2 +1,243 @@
...
+```
+
+And submit this last transaction by moving it to the mempool path:
+
+```bash
+cp ./tx4.bin /tmp/stacks-testnet-5fc814cf78dc0636/L1/mempool
+```
+
+Congratulations, you can now [write your own smart contracts with Clarity](https://docs.blockstack.org/core/smart/overview.html).
+
```
Comment (jcnelson) [inline README.md]: In the tutorial, you might want to talk about the `nonce` argument a little bit so developers know to increment it on each transaction.

File: README.md (hunk 2 above)
Comment (kantai) [reply README.md]: I think it works either way -- the header of repo's readme is usually a "noun" description of what the repo is.

"This is the reference implementation of whitepaper in Rust."

File: README.md [hunk 9]
```diff
@@ -1,2 +1,243 @@
...
+Reference implementation of https://blockstack.org/whitepaper.pdf in Rust.
+
+[![CircleCI](https://circleci.com/gh/blockstack/blockstack-core/tree/develop.svg?style=svg)](https://circleci.com/gh/blockstack/blockstack-core/tree/develop)
+
+## Repository
+
+| Blockstack Topic/Tech | Where to learn more more |
+|---------------------------------|------------------------------------------------------------------------------|
+| Stacks 2.0 | [master branch](https://github.com/blockstack/blockstack-core/tree/master) |
+| Stacks 1.0 | [legacy branch](https://github.com/blockstack/blockstack-core/tree/develop) |
```
Comment (kantai) [inline README.md]: the legacy branch won't be `develop` -- it should be `stacks-1.0` or something along those lines.

File: README.md [hunk 10]
```diff
@@ -1,2 +1,243 @@
...
+
+# Output
+# {
+#  secretKey: "b8d99fd45da58038d630d9855d3ca2466e8e0f89d3894c4724f0efc9ff4b51f001",
+#  publicKey: "02781d2d3a545afdb7f6013a8241b9e400475397516a0d0f76863c6742210539b5",
+#  stacksAddress: "ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH"
+# }
+```
+
+We are considering the following simple contract `kv-store`, that you can save to `./kv-store.clar`:
```
Comment (kantai) [inline README.md]: "We will interact with the following simple contract `kv-store`. In our examples, we will assume this contract is saved to `./kv-store.clar`:"

File: README.md [hunk 11]
```diff
@@ -1,2 +1,243 @@
...
+        entry (ok (get value entry))
+        (err 0)))
+
+(define-public (set-value (key (buff 32)) (value (buff 32)))
+    (begin
+        (map-set store ((key key)) ((value value)))
+        (ok 'true)))
+```
+
+We want to publish this contract on chain, and issuing some transactions interacting with it, setting some keys and getting some values, so we can observe read and writes.
```
Comment (kantai) [inline README.md]: We want to publish this contract on chain, then issue some transactions that interact with it by setting some keys and getting some values, so we can observe read and writes.

File: README.md (hunk 1 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 2 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 3 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 4 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 5 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 6 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 7 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 8 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 9 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 10 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

File: README.md (hunk 11 above)
Comment (lgalabru) [reply README.md]: Addressed with https://github.com/blockstack/blockstack-core/pull/1251/commits/58690348304e8c9c91e76132e25a06db443883d1

**Comment by jcnelson** [comment]
LGTM!

**Pull request merged**